pkg/kopialib/storage/storage_interface.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/pkg/errors"
+)
+
+type StorageType string
+
+const (
+	TypeS3        StorageType = "S3"
+	TypeAzure     StorageType = "Azure"
+	TypeFileStore StorageType = "FileStore"
+	TypeGCP       StorageType = "GCP"
+)
+
+type Storage interface {
+	Connect() (blob.Storage, error)
+	SetOptions(context.Context, map[string]string) error
+	WithCreate(bool)
+}
+
+func New(storageType StorageType) Storage {
+	switch storageType {
+	case TypeS3:
+		return &s3Storage{}
+	default:
+		return nil
+	}
+}
+
+func validateCommonStorageArgs(options map[string]string, storageType StorageType, requiredArgs []string) error {
```

Comment:
if this validates common args for a specific implementation, we should have this as a method in the implementation itself. So that we can call it like

```
s3.validateCommonStorageArgs()
```
or something in the same lines.

Reply:
No it is generic for all, other storage providers would also have it

Reply:
> No it is generic for all, other storage providers would also have it

Sorry, I think I didn't understand properly what you meant by generic here. Do you mean this utility is common and would be used by all implementations?
If not, and other providers should also have it, in that case should we move this to provider's source file. That is s3.go?

Reply:
It is common and used by all the providers.